### ISSUE TITLE
Skaffold sync fails if RBAC does not let user list pods at cluster scope

### ISSUE THREAD

**Issue by alewis001** [GoogleContainerTools/skaffold#1525]
### Expected behavior
`skaffold dev` syncs the files in the pod(s) it is watching without requiring the user to have cluster level access.

### Actual behavior
Please note: This may be related to #1521 but thought I'd raise it as a separate issue just in case it's not as simple as that.

When changing a file that is "sync-able" the following error is reported:

`WARN[0093] Skipping deploy due to sync error: copying files: getting pods: pods is forbidden: User "user@domain.com" cannot list pods at the cluster scope`

I'm using a K8s cluster that is managed by an operations team and as such users are granted access to specific namespaces and are not granted access to list pods at the cluster level. 

### Information

- Skaffold version: v0.21.1
- Operating system: macOS Mojave (10.14.2)
- Contents of skaffold.yaml:

The '***' below is an internal registry so I've omitted it in YAML below.
```yaml
apiVersion: skaffold/v1beta1
kind: Config
build:
  artifacts:
  - image: ***
    context: ./build/docker/
    sync:
      '*': ./build/docker/
  tagPolicy:
    sha256: {}
deploy:
  kubectl:
    manifests:
    - ./helm/myapp.yaml
```

### Steps to reproduce the behavior

1. `skaffold  dev --tail=false --port-forward=false` (The cluster is v1.10.1 so I'm affected by #1181).
2. Modify syncable file (in my case, rebuild a war file).

**Comment by michaelfig** [comment]
I can verify that I had this issue in v0.21.1, and that it is fixed by #1473, which has been merged into master.

I'm just waiting to be able to retest #1521 and this issue against an official release.

**Comment by alewis001** [comment]
Hi. I can confirm that with 0.22.0 I no longer see the error messages so I'll close this issue.

On a separate note, I don't see my updated file (a war) appearing in the running container but I'll do a bit more digging and raise a separate issue if necessary.
